### PR TITLE
Improved encoder M0 settings

### PR DESCRIPTION
**Pull request by NaderMahdi**
** Better speed/quality for M0 (About 70% speed up for similar BDrate)**

** Intra_pred_mode:
 level 4 for base layer pictures and Level 2 otherwise.

**SG-filter level:
  Use gi_step equal to 16

**NSQ_search_Level
	Use level 2;
** Tx Search
    use tx_weight equal to 125.

### CODE REVIEW

**Comment by hassount** [comment]
@NaderMahdi 
lgtm, but can you please fix the linux warnings?

**Comment by NaderMahdi** [comment]
Done.

File: Source/Lib/Codec/EbPictureDecisionProcess.c [hunk 1]
```diff
@@ -791,17 +812,62 @@ EbErrorType signal_derivation_multi_processes_oq(
...
+        break;
+    case 3:
+        intra_pred_level = 2; //ENC_M3
+        break;
+    default:
+        intra_pred_level = 4; //MR_MODE
+        break;
+    }
+
+    if (intra_pred_level == 4) {
```
Comment (kylophone) [inline Source/Lib/Codec/EbPictureDecisionProcess.c]: You could combine all this logic into the switch above. Might be easier to read.

Comment (NaderMahdi) [reply Source/Lib/Codec/EbPictureDecisionProcess.c]: This is just a temporal code to allow the testing team to build their own combination as they wish. I will be fixed once we converge on the intra_pred_levels for all encoder modes.

**Comment by hassount** [comment]
@kylophone 
Can we proceed with this PR ?

**Comment by kylophone** [comment]
> @kylophone
> Can we proceed with this PR ?

OK

**Pull request merged**